Dockerfile

```diff
@@ -0,0 +1,22 @@
+ARG GO_VERSION
+FROM golang:${GO_VERSION}-buster AS builder
```

Comment:
Why are we using this instead of the same image as we base https://github.com/elastic/fleet-server/blob/main/Dockerfile.build off of?

Reply:
Both Dockerfiles have different purpouses, `Dockerfile.build` is used to build the release packages for all platforms, and writes the generated packages to the working copy. This one only builds the binary in the platform it is executed, without writing to the working copy, and from it builds the docker image.

I could make a common Dockerfile for both to import, but I think they wouldn't have so much common code. We would also need to somehow handle the different platforms here.

It would be good though to use here `make release`, to ensure that we are building the same release artifacts in both cases.

I will give another thought to this. I am mainly using this Dockerfile now to have an image for testing on Kubernetes. We could have a more final Dockerfile later.

Reply:
I have replace the build command in the Dockerfile to use `make release-linux/amd64` instead, so we build with the same options as other release builds.

We will have to revisit this if we support other OSs or architectures.